BUG: TypeError when astropy is missing in pycno

If you don't have astropy, pycno errors in a way it should not.
```
    def pycno(
        gdf, value_field, cellsize, r=0.2, handle_null=True, converge=3, verbose=True
    ):
        try:
            from astropy.convolution import convolve as astro_convolve
        except Exception:
>           raise ("Pycnophylactic interpolation requires the astropy package")
E           TypeError: exceptions must derive from BaseException
```

I guess we should do

```py
 try:
    from astropy.convolution import convolve as astro_convolve
except (ImportError, ModuleNotFoundError):
    raise ImportError("Pycnophylactic interpolation requires the astropy package")
```

That way we also pass through any other errors that may be coming from astropy. I'll make a PR.

--- a/tobler/pycno/pycno.py
+++ b/tobler/pycno/pycno.py
@@ -30,8 +30,8 @@ def pycno(
 ):
     try:
         from astropy.convolution import convolve as astro_convolve
-    except Exception:
-        raise ("Pycnophylactic interpolation requires the astropy package")
+    except (ImportError, ModuleNotFoundError):
+        raise ImportError("Pycnophylactic interpolation requires the astropy package")
 
     """Returns a smooth pycnophylactic interpolation raster for a given geodataframe
 
@@ -286,7 +286,7 @@ def pycno_interpolate(
     Original implementation written by @danlewis85 at <https://github.com/danlewis85/pycno/>
     and based in part on the R pycno package by Chris Brusndon (<https://cran.r-project.org/web/packages/pycno/index.html>)
 
-    References: :cite:`tobler_smooth_1979` 
+    References: :cite:`tobler_smooth_1979`
     """
     assert source_df.crs.equals(
         target_df.crs

